refactor: bug report template (#2425)
jazzband/pip-tools: .github/ISSUE_TEMPLATE/bug-report.md

---

name: Bug report

> description: Create a report to help us improve the package.

Reviewer: ```suggestion
description: Create a report to help us improve pip-tools.
```

Shorter and more direct! 😁

Author: Yep agreed!

body:
    - type: textarea
      attributes:
        label: Describe the Bug
        description: >-
            A clear and concise description of what the bug is.
        validation:

>             required: true

Reviewer: When I did my testing, I found that `validation` is not allowed on `textarea` items. I think we were misled by copying from a more complex issue template (the yarl one we used as an example).

```suggestion
```

Author: Hmm, I checked the [docs](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema): 

And it looks like its called `validations`.

It seems like the `validations` key is also only available for public repos, which might be why you couldnt get it to work in your private repo.

This is what i saw about a quarter of the way down:

<img width="755" height="755" alt="image" src="https://github.com/user-attachments/assets/fccc8856-ec1b-4407-b3ff-d9cb4ac3a213" />

I tested it in one of my own public [repo's](https://github.com/psthindal/dev_resources/blob/main/.github/ISSUE_TEMPLATE/bug_report.yaml?plain=1) , and it seems to work as expected.

Let me know if you can recreate this behaviour, if so I think it would be useful to add it into the template for the describe the bug and expected behaviour sections of the bug reporting template.

Author: **[Fixed]** - in commit: <https://github.com/jazzband/pip-tools/commit/2003511546faa1601891d4fcc7e38d7d72a50633>.

Reviewer: Yep! I also tried it out and confirmed it works with `validations` (plural). Good to know!

I also cross-checked using `check-jsonschema`[^1]:
```
$ check-jsonschema --builtin-schema vendor.github-issue-forms .github/ISSUE_TEMPLATE/bug_report.yaml
ok -- validation done
```

[^1]: big disclaimer: I'm the maintainer :grin:

    - type: textarea
      attributes:
        label: Expected Behaviour
        description: >-
           A description of what were you expecting to happen.
        validation:

>             required: true

Reviewer: ```suggestion
```

Author: Please see the above comment and commit: <https://github.com/jazzband/pip-tools/commit/2003511546faa1601891d4fcc7e38d7d72a50633>. Cheers

    - type: textarea
      attributes:
        label: Steps to Reproduce
        description: >-

>             Describe the reproduce to reproduce the bug. 

Reviewer: Typo:
```suggestion
            Describe the steps to reproduce the bug. 
```

        placeholder: |
            1. ...
            2. ...
            3. ...

    - type: textarea
      attributes:
        label: Environment Versions
        description: >-
            Describe the environment versions used 
        placeholder: |
            1. OS Type
            2. Python version: `$ python -V`
            3. pip version: `$ pip --version`

>             4. pip-tools version: `$ pip-compile --version`

Reviewer: I think it works to have this file named `bug-report.md` (GitHub is weird 😉 ), but that we should rename it to `.yaml`, since it's no longer a markdown doc.

Author: Agreed!

Author: **[Fixed]** - in commit: <https://github.com/jazzband/pip-tools/commit/2003511546faa1601891d4fcc7e38d7d72a50633>.
